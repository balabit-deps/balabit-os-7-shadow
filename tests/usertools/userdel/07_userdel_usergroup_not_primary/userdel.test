#!/bin/sh

set -e

cd $(dirname $0)

. ../../../common/config.sh
. ../../../common/log.sh

log_start "$0" "userdel does not remove the user's group if it has a different name"

save_config

# restore the files on exit
trap 'log_status "$0" "FAILURE"; restore_config' 0

change_config

echo -n "Delete user foo (userdel foo)..."
userdel foo 2>tmp/userdel.out
echo "OK"

echo "userdel reported:"
echo "======================================================================="
cat tmp/userdel.out
echo "======================================================================="
echo -n "Check the usage message..."
diff -au data/userdel.out tmp/userdel.out
echo "usage message OK."
rm -f tmp/userdel.out

echo -n "Check the passwd file..."
../../../common/compare_file.pl data/passwd /etc/passwd
echo "OK"
echo -n "Check the group file..."
../../../common/compare_file.pl data/group /etc/group
echo "OK"
echo -n "Check the shadow file..."
../../../common/compare_file.pl data/shadow /etc/shadow
echo "OK"
echo -n "Check the gshadow file..."
../../../common/compare_file.pl data/gshadow /etc/gshadow
echo "OK"

log_status "$0" "SUCCESS"
restore_config
trap '' 0

