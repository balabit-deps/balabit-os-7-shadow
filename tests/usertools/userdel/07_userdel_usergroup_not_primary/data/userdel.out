userdel: group foo not removed because it is not the primary group of user foo.
